src/widget/genericchatroomwidget.cpp

```diff
@@ -72,6 +72,7 @@ void GenericChatroomWidget::compactChange(bool _compact)
     // avatar
     if (isCompact())
     {
+        delete textLayout;  // Not needed
```

Comment:
Argh no, my proposed change will lead to crash (delete dangling pointer on subsequent calls). You would also need to set `textLayout = nullptr;` after line 59. Anyway, I'll happily help to clean this up in a new PR - merging this one later today.